lib/format_linux_xdp.c

```diff
@@ -908,32 +1092,47 @@ static int linux_xdp_read_stream(libtrace_t *libtrace,
     struct pollfd fds;
     int ret;
     uint64_t sys_time;
+    uint64_t release_addr;
 
     if (libtrace->format_data == NULL) {
         trace_set_err(libtrace, TRACE_ERR_BAD_FORMAT, "Trace format data missing, "
             "call trace_create() before calling trace_read_packet()");
         return -1;
     }
 
-    /* check nb_packets (request packets) is not larger than the max RX_BATCH_SIZE */
-    if (nb_packets > RX_BATCH_SIZE) {
-        nb_packets = RX_BATCH_SIZE;
-    }
-
     fds.fd = xsk_socket__fd(stream->xsk->xsk);
     fds.events = POLLIN;
 
     /* try get nb_packets */
     while (rcvd < 1) {
+
+        // check for any addrs to be released back to the fill queue
+        while (libtrace_ringbuffer_try_read(&stream->addr_free_ring, (void **)&release_addr) == 1) {
+            ret = xsk_ring_prod__reserve(&stream->xsk->umem->fq, 1, &idx_rx);
```

Comment:
The looks to the identical code in xdp_fin_packet, can we either call xdp_fin_packet(), or refactor this part out into its own function.